python/ray/air/execution/resources/placement_group.py

```diff
@@ -0,0 +1,216 @@
+import time
+from collections import defaultdict
+from typing import Dict, List, Optional, Set, Union
+
+from dataclasses import dataclass
+
+import ray
+from ray.air.execution.resources.request import (
+    ResourceRequest,
+    AcquiredResource,
+    RemoteRayObject,
+)
+from ray.air.execution.resources.resource_manager import ResourceManager
+from ray.util.annotations import DeveloperAPI
+from ray.util.placement_group import PlacementGroup, remove_placement_group
+from ray.util.scheduling_strategies import PlacementGroupSchedulingStrategy
+
+
+@DeveloperAPI
+@dataclass
+class PlacementGroupAcquiredResource(AcquiredResource):
+    placement_group: PlacementGroup
+
+    def annotate_remote_objects(
+        self, objects: List[RemoteRayObject]
+    ) -> List[Union[RemoteRayObject]]:
+        # If we have an empty head, we schedule the first object (the "head") in the
+        # first PG bundle with num_cpus=0. The second object will also be scheduled in
+        # the first bundle, but will occupy the respective resources. Thus, we start
+        # counting from i = -1, and bundle = max(i, 0) which will be [0, 0, 1, 2, ...]
+        if self.resource_request.head_bundle_is_empty:
+            start = -1
+            bundles = [{}] + self.resource_request.bundles
+        else:
+            start = 0
+            bundles = self.resource_request.bundles
```

Comment:
That's a great suggestion! I've moved the commong logic to the base class so that PGs/fixed only have the backend-specific annotation logic. That should also resolve you initial confusion that the two methods were very similar